### DISCUSSION TITLE
router.push make beforeeach run twice

### DISCUSSION THREAD

**Discussion by mc-tracker** [vuejs/router#1742] [Help and Questions] [answered]
### Reproduction

reproduce easily

### Steps to reproduce the bug

1、declare beforeeach with router and embed console.log  in it for log.
2、use router.push to navigate any path.
3、check the console log on browser.

### Expected behavior

log is printed once after router.push.

### Actual behavior

log is printed twice after router.push.

**Comment by mc-tracker** [comment]
I find stable steps to reproduce. the additioal information is that the element which trigger router.push behavior  is a `<a>`  and it's href is set to '#'. it is binded an click event which call router.push.

**Comment by posva** [comment, marked as answer]
There is also the initial navigation that makes the `router.beforeEach()` run once for the initial navigation (app.use(router) on client). The second log comes from your router.push()

An another possibility is you put the `router.beforeEach()` within a component's `setup`, so every time that component is mounted, a new navigation guard is added without removing the other one when unmounting.

**Reply by mc-tracker** [reply]
no no no，navigation had been initialized when I called route.push.  and I defined beforeeach in router.js rather than setup. when I used router.push to navigate a new page (e.g second.vue)，the console log was printed twice. 

for reminder,In my previous comment on above, this issue only happend on `<a>` tag which was binded click event and call route.push
